automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -0,0 +1,143 @@
+        statusRequest.status shouldBe StatusCodes.OK // compare
+      }
+    }
+
+    "should allow snapshot references to be added to workspaces" taggedAs(Tags.AlphaTest, Tags.ExcludeInFiab) in {
+      // as of this writing, hermione.owner is the user with access to snapshots
+      val owner = UserPool.userConfig.Owners.getUserCredential("hermione")
+
+      implicit val ownerAuthToken: AuthToken = owner.makeAuthToken()
+
+      withCleanBillingProject(owner) { projectName =>
+        withWorkspace(projectName, s"${UUID.randomUUID().toString}-snapshot references") { workspaceName =>
+
+          // call data repo to list snapshots
+          // this gets the most recent snapshots in TDR (to which we have read access). This can cause tests to change
+          // over time, if the snapshots keep changing. It's here for convenience - we can always add/remove snapshots from
+          // TDR and this code will always pick up the latest. But if it causes flakiness we could change to sorting by
+          // created_date ASC - that should be more stable.
+          val dataRepoDAO = new TestDataRepoDAO("terra", dataRepoBaseUrl).getRepositoryApi(ownerAuthToken)
+
+          logger.info(s"calling data repo at $dataRepoBaseUrl as user ${owner.email} ... ")
+          val drSnapshots = Try(dataRepoDAO.enumerateSnapshots(
+            0, 2, "created_date", "desc", "")) match {
+            case Success(s) => s
+            case Failure(ex) =>
+              logger.error(s"data repo call as user ${owner.email} failed: ${ex.getMessage}", ex)
+              throw ex
+          }
+          assume(drSnapshots.getItems.size() == 2,
+            s"TDR at $dataRepoBaseUrl did not have 2 snapshots for this test to use!")
+
+          val dataRepoSnapshotId = drSnapshots.getItems.get(0).getId
+          val anotherDataRepoSnapshotId = drSnapshots.getItems.get(1).getId
+
+          logger.info(s"found 2 snapshots from $dataRepoBaseUrl as user ${owner.email}: $dataRepoSnapshotId, $anotherDataRepoSnapshotId")
+
+          // add snapshot reference to the workspace. Under the covers, this creates the workspace in WSM and adds the ref
+          createSnapshotReference(projectName, workspaceName, dataRepoSnapshotId, "firstSnapshot")
+
+          // validate the snapshot was added correctly: list snapshots in Rawls, should return 1, which we just added.
+          // if we can successfully list snapshot references, it means WSM created its copy of the workspace
+          val firstListResponse = listSnapshotReferences(projectName, workspaceName)
+          val firstResources = Rawls.parseResponseAs[DataReferenceList](firstListResponse).getResources.asScala
+          firstResources.size shouldBe 1
+          firstResources.head.getName shouldBe "firstSnapshot"
+          firstResources.head.getReferenceType shouldBe ReferenceTypeEnum.DATA_REPO_SNAPSHOT
+          firstResources.head.getReference.getSnapshot shouldBe dataRepoSnapshotId
```

Comment:
Is it worth verifying snapshot tables and details? That's part of the proposal, although was not meant to be prescriptive. https://docs.google.com/document/d/19LpGtamwKC9SsuwU7WfXgLngI6qR7NSdwvCwhaQfCjo/edit#heading=h.51nwpgackk53

Reply:
My gut says that the most likely failure point is in creating the references. That said ... it's worth writing a test to verify  tables. I propose doing this in a separate test and as a separate PR to keep things small and contained,  how does that sound?

Reply:
That sounds good to me.